main.tf

```diff
@@ -99,6 +162,13 @@ resource "ibm_is_subnet_reserved_ip" "vsi_ip" {
   auto_delete = false
 }
 
+resource "ibm_is_subnet_reserved_ip" "secondary_vsi_ip" {
+  for_each    = var.number_of_secondary_reserved_ips > 0 && !var.use_legacy_network_interface ? local.secondary_reserved_ips_map : {}
```

Comment:
see the for_each in the vsi_ip above for how to avoid a bug in terraform with a conditional returning map